src/Storages/FileLog/FileLogDirectoryWatcher.cpp

```diff
@@ -0,0 +1,88 @@
+#include <Storages/FileLog/FileLogDirectoryWatcher.h>
+#include <Poco/Delegate.h>
+#include <Poco/DirectoryWatcher.h>
+
+FileLogDirectoryWatcher::FileLogDirectoryWatcher(const std::string & path_)
+    : path(path_), dw(std::make_shared<Poco::DirectoryWatcher>(path))
+{
+    dw->itemAdded += Poco::delegate(this, &FileLogDirectoryWatcher::onItemAdded);
+    dw->itemRemoved += Poco::delegate(this, &FileLogDirectoryWatcher::onItemRemoved);
+    dw->itemModified += Poco::delegate(this, &FileLogDirectoryWatcher::onItemModified);
+    dw->itemMovedFrom += Poco::delegate(this, &FileLogDirectoryWatcher::onItemMovedFrom);
+    dw->itemMovedTo += Poco::delegate(this, &FileLogDirectoryWatcher::onItemMovedTo);
```

Comment:
May be worth adding a comment from poco documentation:
`DW_ITEM_MOVED_FROM and DW_ITEM_MOVED_TO events will only be reported on Linux. On other platforms, a file rename or move operation will be reported via a DW_ITEM_REMOVED and a DW_ITEM_ADDED event. The order of these two events is not defined.`

Reply:
OK!